src/App.tsx

```diff
@@ -17,18 +50,30 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                onStatusChange={setStatus}
+                onQueryChange={setQuery}
+                query={query}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {todosFromServer.length === 0 && <Loader />}
+              {todosFromServer.length !== 0 && (
```

Comment:
would be nice to have one line indent to make neighbouring condition statements more readable